test/tasks/publish/test_local.py

```diff
@@ -29,7 +29,10 @@ def test_execute(self, exec_command):
         task = local.LocalStagingTask()
         task.execute(
             gapic_code_dir=os.path.expanduser('~/foo/bar'),
-            git_repo={'gapic_subpath': 'pubsub'},
+            git_repo={
+                'gapic_subpath': 'pubsub',
+                'location': 'api_client_staging.git',
```

Comment:
Just checked: You can have hyphens, but you have to quote the key. I feel that underscores are more readable here.

That said, the spot that you identified is actually an error, and those should be hyphens. (It is not relevant because it is a unit test and the bogus data does not matter; I could have named it foobar.git, but I would like it to match.)

For overriding in `local_paths`, though, it would be underscores. (That said, I expect most people will not override this at all and just let it derive from `reporoot`.)